talk/expert/sfinae.tex

```diff
@@ -218,55 +249,54 @@
         In& operator()(In* in) const {
           assert(in!=nullptr); return *in;
         }
-      } deref{};
+      } deref;
     \end{cppcode*}
   \end{exampleblock}
 \end{frame}
 
 \begin{frame}[fragile]
-  \frametitlecpp[11]{Back to variadic templated class}
-  \begin{block}{The tuple get method}
+  \frametitlecpp[11]{Back to variadic class templates}
+  \begin{block}{The \texttt{tuple\_element} trait}
     \begin{cppcode*}{}
       template <size_t I, typename Tuple>
-      struct elem_type;
+      struct tuple_element;
```

Comment:
I do not care so much about the name, but just as a historical note : originally this was really a cut and paste from the standard library

Reply:
Really? The type function is called [`tuple_element`](https://en.cppreference.com/w/cpp/utility/tuple/tuple_element) nowadays. I did not know it had a different name back then! :D

Reply:
I confirm the latest code of the gcc implementation uses tuple_element